code/parachain/frame/composable-traits/src/dex.rs

```diff
@@ -383,3 +389,132 @@ mod tests {
 		);
 	}
 }
+
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Default, PartialEq, Eq, Copy, RuntimeDebug,
+)]
+pub struct AssetDepositInfo<AssetId> {
+	pub asset_id: AssetId,
+	pub deposit_amount: u128,
+	// rename to `existing_balance`
+	pub asset_balance: u128,
+	// could probably just be called `weight`?
+	pub asset_weight: Permill,
+}
+
+impl<AssetId> AssetDepositInfo<AssetId> {
+	pub fn from(deposit_amount: AssetAmount<AssetId, u128>, asset_info: (Permill, u128)) -> Self {
+		// TODO(ben,connor): debug assert asset_balance is non-zero?
+		AssetDepositInfo {
+			asset_id: deposit_amount.asset_id,
+			deposit_amount: deposit_amount.amount,
+			asset_balance: asset_info.1,
+			asset_weight: asset_info.0,
+		}
+	}
+
+	pub fn get_deposit_ratio(&self) -> Rational128 {
+		Rational128::from(self.deposit_amount, self.asset_balance)
+	}
+
+	pub fn cmp_by_deposit_ratio(&self, other: Self) -> Ordering {
+		self.get_deposit_ratio().cmp(&other.get_deposit_ratio())
+	}
+}
+
+pub fn normalize_asset_deposit_infos_to_min_ratio<AssetId: Copy>(
```

Comment:
Need a comment on the algo used

Reply:
added a simple doc comment, not sure what else to add